packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -235,13 +295,54 @@ class IosProject extends XcodeBasedProject {
     // only used for display purposes and to regenerate organization names, so
     // best-effort is probably fine.
     final String? fromPbxproj = firstMatchInFile(xcodeProjectInfoFile, _productBundleIdPattern)?.group(2);
-    if (fromPbxproj != null && (fromPlist == null || fromPlist == _productBundleIdVariable)) {
+    if (fromPbxproj != null && (fromPlist == null || fromPlist == _kProductBundleIdVariable)) {
       return fromPbxproj;
     }
+    return null;
+  }
 
+  Future<String?> _getTeamIdentifier(XcodeProjectBuildContext buildContext) async {
+    final Map<String, String>? buildSettings = await _buildSettingsForXcodeProjectBuildContext(buildContext);
+    if (buildSettings != null) {
+      return buildSettings[kTeamIdKey];
+    }
     return null;
   }
 
+  Future<List<String>> _getAssociatedDomains(XcodeProjectBuildContext buildContext) async {
+    final Map<String, String>? buildSettings = await _buildSettingsForXcodeProjectBuildContext(buildContext);
+    if (buildSettings != null) {
+      final String? entitlementPath = buildSettings[kEntitlementFilePathKey];
+      if (entitlementPath != null) {
+        final File entitlement = hostAppRoot.childFile(entitlementPath);
+        if (entitlement.existsSync()) {
+          List<String>? domains;
+          try {
+            final List<Object>? value = globals.plistParser.getValueFromFile<List<Object>>(
+              entitlement.path,
+              PlistParser.kAssociatedDomainsKey,
+            );
+            domains = value?.cast<String>();
+          } on FileNotFoundException {
+            // iOS tooling not found; likely not running OSX;
+          }
```

Comment:
I don't think we would able to get the build settings if we weren't running on macOS, it seems like this should maybe be a `toolExit()` because the entitlements file path isn't set up correctly?

Reply:
Oh wait, you do. Then why would we hit `FileNotFoundException`?

Reply:
good call